openedx/core/djangoapps/notifications/views.py

```diff
@@ -219,7 +219,8 @@ def patch(self, request, course_key_string):
         updated_notification_preferences = preference_update.save()
         notification_preference_update_event(request.user, course_id, preference_update.validated_data)
         serializer = UserCourseNotificationPreferenceSerializer(updated_notification_preferences)
-        return Response(serializer.data, status=status.HTTP_200_OK)
+        preferences = filter_course_wide_preferences(course_id, serializer.data)
```

Comment:
I think it would make sense to update preferences and then pass it to the serializer?

Reply:
`updated_notification_preferences` is a model instance. I think it is not a good idea to update any model object until it needs to be updated.

Reply:
We are also using similar logic in `GET` call